### PR TITLE
feat: handle multiple tags filter on datasets

### PR DESCRIPTION
**Pull request by abulte**
Same as https://github.com/opendatateam/udata/pull/3204 but for search service.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ dependencies = [
     "pytest-flask==1.2.0",
     "markdown==3.3.3",
     "beautifulsoup4==4.10.0",
+    "werkzeug==3.0.4",
```
Comment (abulte) [inline pyproject.toml:25]: 3.1.3 will fail during tests with

```
__________ ERROR at setup of test_api_reuse_index_on_another_index ___________

app = <Flask 'udata_search_service.app'>

    @pytest.fixture
    def client(app):
>       with app.test_client() as client:

tests/conftest.py:17:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pyenv/lib/python3.11/site-packages/flask/app.py:1252: in test_client
    return cls(  # type: ignore
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <FlaskClient <Flask 'udata_search_service.app'>>
args = (<Flask 'udata_search_service.app'>, <class 'pytest_flask.plugin.<class 'pytest_flask.plugin.JSONResponse'>'>)
kwargs = {'use_cookies': True}

    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
        super().__init__(*args, **kwargs)
        self.preserve_context = False
        self._new_contexts: t.List[t.ContextManager[t.Any]] = []
        self._context_stack = ExitStack()
        self.environ_base = {
            "REMOTE_ADDR": "127.0.0.1",
>           "HTTP_USER_AGENT": f"werkzeug/{werkzeug.__version__}",
        }
E       AttributeError: module 'werkzeug' has no attribute '__version__'

pyenv/lib/python3.11/site-packages/flask/testing.py:116: AttributeError
```

File: udata_search_service/presentation/api.py [hunk 2]
```diff
@@ -69,6 +70,22 @@ def sort_validate(cls, value):
...
+                if key in cls.__fields__
+                and is_list_type(cls.__fields__[key].outer_type_)
+            }
+
+        return cls(
+            **{
+                **request_args.to_dict(),
+                **get_list_args(),
+            }
+        )
```
Comment (abulte) [inline udata_search_service/presentation/api.py:88]: Inspired by https://github.com/bauerji/flask-pydantic/blob/8595fa8b5513a336c9c679829f49ddc20f56377d/flask_pydantic/converters.py#L21

File: udata_search_service/infrastructure/search_clients.py [hunk 3]
```diff
@@ -258,6 +258,12 @@ def query_datasets(self, query_text: str, offset: int, page_size: int, filters:
                 search = search.filter('range', **{'temporal_coverage_start': {'lte': value}})
             elif key == 'temporal_coverage_end':
                 search = search.filter('range', **{'temporal_coverage_end': {'gte': value}})
+            elif key == 'tags':
```
[the remark this answers is not in the document]
Comment (abulte) [reply udata_search_service/infrastructure/search_clients.py:261]: `tag` is converted beforehand to `tags` https://github.com/opendatateam/udata-search-service/blob/0c8a1f7ec807db224b17872f19451dc4fbb2bb97/udata_search_service/infrastructure/services.py#L90, and no change has been introduced about that in this PR. Seems clean enough to me.

[the remark this answers is not in the document]
Comment (magopian) [reply udata_search_service/infrastructure/search_clients.py:261]: Ah, yes indeed, good point. Thanks for the clarification!

**Review by maudetes — APPROVED** [review]
Thank you for this PR as well! :clap:

File: tests/test_search_client.py [hunk 4]
```diff
@@ -190,7 +190,7 @@ def test_search_datasets_with_temporal_filters(app, client, search_client, faker
 def test_search_with_tag_filter(app, client, search_client, faker):
     for i in range(4):
         search_client.index_dataset(DatasetFactory(
-            tags=['test-tag'] if i % 2 else ['not-test-tag']
+            tags=['test-tag', 'test-tag-2'] if i % 2 else ['not-test-tag']
```
[the remark this answers is not in the document]
Comment (abulte) [reply tests/test_search_client.py]: Changed in https://github.com/opendatateam/udata-search-service/pull/50/commits/3a176c31c45fa215ca8ecfb02dc61209502d912d. Is this what you want? I'm not sure it's testing more cases 🤔

**Review by magopian — APPROVED** [review]
Nice!

File: tests/test_search_client.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (maudetes) [reply tests/test_search_client.py]: I may be be missing somthing but I think that the results before were the same between
```
results_number, res = search_client.query_datasets(None, 0, 20, {'tags': ['test-tag', 'test-tag-2']})
```
and
```
results_number, res = search_client.query_datasets(None, 0, 20, {'tags': ['test-tag']})
```
thus, not testing the double tags filter?

**Pull request merged**